angucomplete-alt.js

```diff
@@ -298,20 +298,35 @@
         }
       }
 
+      function handleSubmitOnEnter(event) {
+        if (event && !scope.submitOnEnter) {
```

Comment:
Nice catch. I think what happens is that since @ is a one-way binding, `scope.submitOnEnter` was being reset each time the scope got applied or something. I declared a separate `submitOnEnter` variable and assigned the value to it instead, which is what is being done for all of the other config options.